src/modules/client/pages/ni/billing/CreditNotes.vue

```diff
@@ -201,25 +206,54 @@ export default {
     const creditNoteValidation = {
       date: { required },
       customer: { required },
-      startDate: { required: requiredIf(() => this.hasLinkedEvents) },
-      endDate: { required: requiredIf(() => this.hasLinkedEvents) },
       events: { required: requiredIf(() => this.hasLinkedEvents) },
       subscription: { required: requiredIf(() => !this.hasLinkedEvents) },
       inclTaxesTpp: {},
       inclTaxesCustomer: {},
     };
-
+    const newCreditNoteDateValidation = {
+      startDate: {
+        required: requiredIf(() => this.hasLinkedEvents),
+        maxDate: this.isValidDate(this.creationMinAndMaxDates.maxStartDate)
+          ? maxDate(this.creationMinAndMaxDates.maxStartDate)
+          : '',
+      },
+      endDate: {
+        required: requiredIf(() => this.hasLinkedEvents),
+        minDate: minDate(this.isValidDate(this.creationMinAndMaxDates.minEndDate)
+          ? this.creationMinAndMaxDates.minEndDate
+          : this.newCreditNote.startDate),
+      },
+    };
+    const editedCreditNoteDateValidation = {
+      startDate: {
+        required: requiredIf(() => this.hasLinkedEvents),
+        maxDate: this.isValidDate(this.editionMinAndMaxDates.maxStartDate)
+          ? maxDate(this.editionMinAndMaxDates.maxStartDate)
+          : '',
+      },
+      endDate: {
+        required: requiredIf(() => this.hasLinkedEvents),
+        minDate: minDate(this.isValidDate(this.editionMinAndMaxDates.minEndDate)
+          ? this.editionMinAndMaxDates.minEndDate
+          : this.editedCreditNote.startDate),
+      },
+    };
```

Comment:
tu peux faire une methode pour mutualiser les 2 en passant en argument `this.newCreditNote` et `this.creationMinAndMaxDates`

Reply:
done